add utility to generate warm-start files for MOM6 and CICE6; modify cpld_gridgen as required by new ocnice_prep utility (was #945)

DeniseWorthen opened the pull request:
## DESCRIPTION OF CHANGES: 
Adds a new utility to down-scale MOM6 and CICE6 restarts from 1/4 deg tripole to a lower resolution, where they can be used as warm-start files.

## TESTS CONDUCTED: 
If there are changes to the build or source code, the tests below must be conducted. Contact a repository manager if you need assistance.

- [ ] Compile branch on Hera using GNU. Currently, the head of 'develop' has a problem with the gnu build module. So, this test is being waived. There is no problem with the branch.
- [x] Compile branch in 'Debug' mode on WCOSS2. Done on Dogwood using cd7ba2d.
- [x] Run unit tests locally on any Tier 1 machine. Done on Dogwood using cd7ba2d.
- [x] Run `cpld_gridgen` and `ocnice_prep` consistency tests locally on all Tier 1 machines. Done on Orion using d07c282. Done on Dogwood, Hera, Jet and Hercules using cd7ba2d.

Describe any additional tests performed.

## DEPENDENCIES:

A unit test testing the expected namelist and fields expected for ocean and ice has been added. 
A simple regression test has been added which utilizes a staged copy of 1/4 deg ocean and ice ICs. These were obtained from the replay ICs staged on AWS ``https://noaa-ufs-gefsv13replay-pds.s3.amazonaws.com/2021/03/2021032206``. The ice restart is renamed generically as ``ice.nc``. The MOM restarts have been pre-processed using the following two NCO commands

```
ncks -v Temp,Salt,h,u MOM.res.nc ocean.nc
ncks -v v,sfc -A MOM.res_1.nc ocean.nc
```

- requires #943 

## DOCUMENTATION:

The code has been doxygenated and builds succesfully on hera. A basic description of the utility has also been added in the ``sorc/ocnice_prep.fd/docs``. 

Documentation for ``cpld_gridgen`` has also been updated to include a figure showing the tripole grid and to clarify the two types of weights produce: postitional weights (e.g. Cu->Ct) and mapping weights (e.g. tripole 1/4deg->rect 1/4 deg)

## ISSUE: 
- fixes #929 
- fixes #943

DeniseWorthen committed:
add wts for ct->cu,cv for ocnprep

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/ocnprep

DeniseWorthen committed:
initial commit of ocean-ice prep

DeniseWorthen committed:
shorten lines for readability

DeniseWorthen committed:
stash work

DeniseWorthen committed:
fix for hmin parameter

DeniseWorthen committed:
initial doxygenation

DeniseWorthen committed:
add Ct->Cu,Cv for generation of ocean warmstarts

DeniseWorthen committed:
documentation builds successfully

DeniseWorthen committed:
update utility description for docs

DeniseWorthen committed:
restore logs and tidy up

DeniseWorthen committed:
remove unused variables

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/ocnprep

DeniseWorthen committed:
commit non-working unit test for ocean ice prep

DeniseWorthen committed:
update test

DeniseWorthen committed:
init_mod testing

DeniseWorthen committed:
mods for nml testing

DeniseWorthen committed:
update nmls for test

DeniseWorthen committed:
remove nml

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/addoitest

DeniseWorthen committed:
Merge remote-tracking branch 'origin/feature/addoitest' into feature/ocnprep

DeniseWorthen committed:
add unit test for valid nml settings

DeniseWorthen committed:
extend unit test for invalid csv values

DeniseWorthen committed:
minor tweaks to documentation

DeniseWorthen committed:
minor tweaks

DeniseWorthen commented:
@GeorgeGayno-NOAA I've added documentation and unit test. The test checks for invalid settings in either the nml or csv file that the code uses. At this point I need to start thinking about how to add a regression test for this utility. Minsuk created the testing framework for cpld_gridgen for me---this isn't really my expertise. Do you have any suggestions for how to proceed w/ adding a regression testing framework?

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA I've added documentation and unit test. The test checks for invalid settings in either the nml or csv file that the code uses. At this point I need to start thinking about how to add a regression test for this utility. Minsuk created the testing framework for cpld_gridgen for me---this isn't really my expertise. Do you have any suggestions for how to proceed w/ adding a regression testing framework?

I can create the regression testing framework. Do you have the case(s) you want to run? Point me to the input data and a sample script.

DeniseWorthen committed:
shorten some lines and add more documentation

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/wts4prep

DeniseWorthen committed:
move weight creation outside of if block

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/wts4prep

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/ocnprep

DeniseWorthen committed:
resolve doc build error

DeniseWorthen committed:
clarify log message for rotremap

DeniseWorthen committed:
update with new required weights

baselines fail with the following missing files. All are expected
The Ct.mx025_SCRIP files are missing because the dependency has
been removed. The Ct->Bu file for mx025 is removed because that
mapping is never required. The tripole:tripole files are removed
because NCL method is deprecated.

28:Comparing tripole.mx025.Ct.to.Bu.bilinear.nc........MISSING file
53:Comparing Ct.mx025_SCRIP.nc........MISSING file
65:Comparing tripole.mx025.Ct.to.mx050.Ct.bilinear.nc........MISSING file
66:Comparing tripole.mx025.Ct.to.mx050.Ct.neareststod.nc........MISSING file
91:Comparing Ct.mx025_SCRIP.nc........MISSING file
102:Comparing tripole.mx025.Ct.to.mx100.Ct.bilinear.nc........MISSING file
103:Comparing tripole.mx025.Ct.to.mx100.Ct.neareststod.nc........MISSING file
127:Comparing Ct.mx025_SCRIP.nc........MISSING file
137:Comparing tripole.mx025.Ct.to.mx500.Ct.bilinear.nc........MISSING file
138:Comparing tripole.mx025.Ct.to.mx500.Ct.neareststod.nc........MISSING file

DeniseWorthen committed:
remove accidental ocnicprep files

DeniseWorthen committed:
update documentation and try to add figure

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/wts4prep

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/ocnprep

DeniseWorthen committed:
revert temporary build changes

DeniseWorthen committed:
update documentation to add figure for tripole grid

DeniseWorthen committed:
add to unit test to check for 5deg destination

* 5deg tripole has only 25 vertical levels. The 5deg RT
does run if modifications are made to use 75 levels, but
since 5deg restarts are avail from the marine DA, remapping
of the 1/4deg restarts will not be supported

DeniseWorthen committed:
remove tabs from a test file

DeniseWorthen committed:
Merge remote-tracking branch 'origin/feature/wts4prep' into feature/ocnprep

DeniseWorthen committed:
commit simple script setup

DeniseWorthen committed:
update script comments

DeniseWorthen commented:
@GeorgeGayno-NOAA I have a simple script set up to run the ``oiprep`` utility. All the required files are all in the ``ocnice_prep.fd`` and committed to the branch for now:

```
ocean.csv : control file for ocean restart fields
ice.csv : control file for ice restart fields 
ociceprep.nml.IN : input nml file
ociceprep.sh : simple execution script
```

I've also merged in the changes I'll need in cpld_gridgen to the branch. I'm assuming that will be merged first.

I need two restart files for the RT as input, ``ocean.nc`` and ``ice.nc``. I have them staged here ``/scratch1/NCEPDEV/nems/Denise.Worthen/WORK/utils_dw/sorc/ocnice_prep.fd``

I think for an RT test, we can always use the same input and verification would be that the downscaled warmstart files are compared. I think running the 1/4->1deg case is probably sufficient?

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA I have a simple script set up to run the `oiprep` utility. All the required files are all in the `ocnice_prep.fd` and committed to the branch for now:
> 
> ```
> ocean.csv : control file for ocean restart fields
> ice.csv : control file for ice restart fields 
> ociceprep.nml.IN : input nml file
> ociceprep.sh : simple execution script
> ```
> 

We might consider placing that under the util subdirectory, where there are scripts for running other programs:
https://github.com/DeniseWorthen/UFS_UTILS/tree/feature/ocnprep/util

> I've also merged in the changes I'll need in cpld_gridgen to the branch. I'm assuming that will be merged first.
> 
> I need two restart files for the RT as input, `ocean.nc` and `ice.nc`. I have them staged here `/scratch1/NCEPDEV/nems/Denise.Worthen/WORK/utils_dw/sorc/ocnice_prep.fd`

I can place these somewhere under the 'role' account directories - `/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/`
> 
> I think for an RT test, we can always use the same input and verification would be that the downscaled warmstart files are compared. I think running the 1/4->1deg case is probably sufficient?

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/wts4prep

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/ocnprep

DeniseWorthen committed:
minor doc tweaks

DeniseWorthen committed:
rudimentary rt test

DeniseWorthen committed:
working rt.sh

DeniseWorthen committed:
minor mod of rt.sh

DeniseWorthen commented:
@GeorgeGayno-NOAA I've built a simple RT test for this utility, so I think I'm close to moving this out of 'draft' status. A couple of questions:

1) The new utility requires a couple of additional weights files. Those will be added in #945. No existing weights will change, but some new ones will be added and a couple will be removed. That means the baseline change because of missing files only. I could combine the two PRs (this and 945), but haven't because they are really two different changes. What do you think?
 
2) The new utility also requires the staging of the 1/4deg ocean and ice restarts. In my rt.sh, I've assumed those will be found and copied from a ``..../ufs_utils/reg_tests/ocnice_prep/input_data`` on each platform. So I need to get them copied there. They currently reside on hera in 

```
/scratch1/NCEPDEV/nems/Denise.Worthen/WORK/addoitest/sorc/ocnice_prep.fd/replay-2021032206/ocean.nc
/scratch1/NCEPDEV/nems/Denise.Worthen/WORK/addoitest/sorc/ocnice_prep.fd/replay-2021032206/ice.nc
```

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA I've built a simple RT test for this utility, so I think I'm close to moving this out of 'draft' status. A couple of questions:
> 
> 1. The new utility requires a couple of additional weights files. Those will be added in [modify cpld_gridgen as required by new ocnice_prep utility #945](https://github.com/ufs-community/UFS_UTILS/pull/945). No existing weights will change, but some new ones will be added and a couple will be removed. That means the baseline change because of missing files only. I could combine the two PRs (this and 945), but haven't because they are really two different changes. What do you think?

Where are the new files? I will stage them in the baseline directory on Hera. Once everything is working, I will stage them on all machines.

> 2. The new utility also requires the staging of the 1/4deg ocean and ice restarts. In my rt.sh, I've assumed those will be found and copied from a `..../ufs_utils/reg_tests/ocnice_prep/input_data` on each platform. So I need to get them copied there. They currently reside on hera in
> 
> ```
> /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/addoitest/sorc/ocnice_prep.fd/replay-2021032206/ocean.nc
> /scratch1/NCEPDEV/nems/Denise.Worthen/WORK/addoitest/sorc/ocnice_prep.fd/replay-2021032206/ice.nc
> ```

Will do that today.

DeniseWorthen commented:
I have a cpld_gridgen run from PR 945 w/ the new weights generated, is it OK to just add them to an existing baseline? 

The reg_test I ran would be ``/scratch1/NCEPDEV/stmp4/Denise.Worthen/CPLD_GRIDGEN/rt_3313950/``.

GeorgeGayno-NOAA commented:
> I have a cpld_gridgen run from PR 945 w/ the new weights generated, is it OK to just add them to an existing baseline?
> 
> The reg_test I ran would be `/scratch1/NCEPDEV/stmp4/Denise.Worthen/CPLD_GRIDGEN/rt_3313950/`.

I added the new weights to the official baseline:

```
/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/cpld_gridgen/baseline_data/100
[role.ufsutils@hfe09 100]$ ls -l tripole.mx100.Ct*
lrwxrwxrwx 1 role.ufsutils nems       40 May 31 19:28 tripole.mx100.Ct.to.Cu.bilinear.nc -> tripole.mx100.Ct.to.Cu.bilinear.nc.pr942
-rw-r--r-- 1 role.ufsutils nems 30414828 May 31 19:28 tripole.mx100.Ct.to.Cu.bilinear.nc.pr942
lrwxrwxrwx 1 role.ufsutils nems       40 May 31 19:38 tripole.mx100.Ct.to.Cv.bilinear.nc -> tripole.mx100.Ct.to.Cv.bilinear.nc.pr942
-rw-r--r-- 1 role.ufsutils nems 32465388 May 31 19:38 tripole.mx100.Ct.to.Cv.bilinear.nc.pr942
```
and 

```
/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/cpld_gridgen/baseline_data/050
[role.ufsutils@hfe09 050]$ ls -l tripole.mx050.Ct*
lrwxrwxrwx 1 role.ufsutils nems        40 May 31 19:07 tripole.mx050.Ct.to.Cu.bilinear.nc -> tripole.mx050.Ct.to.Cu.bilinear.nc.pr942
-rw-r--r-- 1 role.ufsutils nems 109488108 May 31 19:05 tripole.mx050.Ct.to.Cu.bilinear.nc.pr942
lrwxrwxrwx 1 role.ufsutils nems        40 May 31 19:16 tripole.mx050.Ct.to.Cv.bilinear.nc -> tripole.mx050.Ct.to.Cv.bilinear.nc.pr942
-rw-r--r-- 1 role.ufsutils nems 117736428 May 31 19:16 tripole.mx050.Ct.to.Cv.bilinear.nc.pr942
```

Let me know if I forgot anything.

GeorgeGayno-NOAA commented:
I also added the input test files to the official directories:

```
/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/ocnice_prep/input_data
[role.ufsutils@hfe09 input_data]$ ls -l
total 2691688
-r--r--r-- 1 role.ufsutils nems  361250167 May 31 15:27 ice.nc
-r--r--r-- 1 role.ufsutils nems 2394969750 May 31 15:27 ocean.nc
```

And I used the output files from my test as 'baseline' files, just to get things to run:

```
/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/ocnice_prep/baseline_data
[role.ufsutils@hfe09 baseline_data]$ ls -lR
.:
total 16
drwxr-sr-x 2 role.ufsutils nems 4096 May 31 17:30 050_ice
drwxr-sr-x 2 role.ufsutils nems 4096 May 31 17:29 050_ocean
drwxr-sr-x 2 role.ufsutils nems 4096 May 31 17:28 100_ice
drwxr-sr-x 2 role.ufsutils nems 4096 May 31 17:26 100_ocean

./050_ice:
total 515180
-rw-r--r-- 1 role.ufsutils nems 527526388 May 31 17:30 ice.mx050.nc

./050_ocean:
total 1464500
-rw-r--r-- 1 role.ufsutils nems 1499629152 May 31 17:29 ocean.mx050.nc

./100_ice:
total 143104
-rw-r--r-- 1 role.ufsutils nems 146536948 May 31 17:28 ice.mx100.nc

./100_ocean:
total 406816
-rw-r--r-- 1 role.ufsutils nems 416564832 May 31 17:26 ocean.mx100.nc
```

DeniseWorthen committed:
removed unneeded files

DeniseWorthen committed:
remove gitignore

DeniseWorthen committed:
update rt.sh for std locations

* add directory slash when reading in directories

DeniseWorthen commented:
@GeorgeGayno-NOAA I updated the ``rt.sh`` to point to the standard locations for both the new weights and the input_data. Everything seems to be working fine.

This PR contains the code changes and RT changes in #945. Do we want to just close that?

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA I updated the `rt.sh` to point to the standard locations for both the new weights and the input_data. Everything seems to be working fine.
> 
> This PR contains the code changes and RT changes in #945. Do we want to just close that?

If #945 is no longer needed, go ahead and close it (and its corresponding issue).  And write a brief note about why it was closed.

GeorgeGayno-NOAA commented:
> > I have a cpld_gridgen run from PR 945 w/ the new weights generated, is it OK to just add them to an existing baseline?
> > The reg_test I ran would be `/scratch1/NCEPDEV/stmp4/Denise.Worthen/CPLD_GRIDGEN/rt_3313950/`.
> 
> I added the new weights to the official baseline:
> 
> ```
> /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/cpld_gridgen/baseline_data/100
> [role.ufsutils@hfe09 100]$ ls -l tripole.mx100.Ct*
> lrwxrwxrwx 1 role.ufsutils nems       40 May 31 19:28 tripole.mx100.Ct.to.Cu.bilinear.nc -> tripole.mx100.Ct.to.Cu.bilinear.nc.pr942
> -rw-r--r-- 1 role.ufsutils nems 30414828 May 31 19:28 tripole.mx100.Ct.to.Cu.bilinear.nc.pr942
> lrwxrwxrwx 1 role.ufsutils nems       40 May 31 19:38 tripole.mx100.Ct.to.Cv.bilinear.nc -> tripole.mx100.Ct.to.Cv.bilinear.nc.pr942
> -rw-r--r-- 1 role.ufsutils nems 32465388 May 31 19:38 tripole.mx100.Ct.to.Cv.bilinear.nc.pr942
> ```
> 
> and
> 
> ```
> /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/cpld_gridgen/baseline_data/050
> [role.ufsutils@hfe09 050]$ ls -l tripole.mx050.Ct*
> lrwxrwxrwx 1 role.ufsutils nems        40 May 31 19:07 tripole.mx050.Ct.to.Cu.bilinear.nc -> tripole.mx050.Ct.to.Cu.bilinear.nc.pr942
> -rw-r--r-- 1 role.ufsutils nems 109488108 May 31 19:05 tripole.mx050.Ct.to.Cu.bilinear.nc.pr942
> lrwxrwxrwx 1 role.ufsutils nems        40 May 31 19:16 tripole.mx050.Ct.to.Cv.bilinear.nc -> tripole.mx050.Ct.to.Cv.bilinear.nc.pr942
> -rw-r--r-- 1 role.ufsutils nems 117736428 May 31 19:16 tripole.mx050.Ct.to.Cv.bilinear.nc.pr942
> ```
> 
> Let me know if I forgot anything.

I can update on each machine. Do you want to create machine-specific versions?

GeorgeGayno-NOAA commented:
> I also added the input test files to the official directories:
> 
> ```
> /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/ocnice_prep/input_data
> [role.ufsutils@hfe09 input_data]$ ls -l
> total 2691688
> -r--r--r-- 1 role.ufsutils nems  361250167 May 31 15:27 ice.nc
> -r--r--r-- 1 role.ufsutils nems 2394969750 May 31 15:27 ocean.nc
> ```
> 
Can we use the same input files on all machines?
> And I used the output files from my test as 'baseline' files, just to get things to run:
> 
> ```
> /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/ocnice_prep/baseline_data
> [role.ufsutils@hfe09 baseline_data]$ ls -lR
> .:
> total 16
> drwxr-sr-x 2 role.ufsutils nems 4096 May 31 17:30 050_ice
> drwxr-sr-x 2 role.ufsutils nems 4096 May 31 17:29 050_ocean
> drwxr-sr-x 2 role.ufsutils nems 4096 May 31 17:28 100_ice
> drwxr-sr-x 2 role.ufsutils nems 4096 May 31 17:26 100_ocean
> 
> ./050_ice:
> total 515180
> -rw-r--r-- 1 role.ufsutils nems 527526388 May 31 17:30 ice.mx050.nc
> 
> ./050_ocean:
> total 1464500
> -rw-r--r-- 1 role.ufsutils nems 1499629152 May 31 17:29 ocean.mx050.nc
> 
> ./100_ice:
> total 143104
> -rw-r--r-- 1 role.ufsutils nems 146536948 May 31 17:28 ice.mx100.nc
> 
> ./100_ocean:
> total 406816
> -rw-r--r-- 1 role.ufsutils nems 416564832 May 31 17:26 ocean.mx100.nc
> ```

If you create versions for each machine, I can place them in the official location.

DeniseWorthen commented:
@GeorgeGayno-NOAA I don't actually know how different any of these weights files might be from platform to platform. Since we do otherwise generate the weights for each platform, I think we should do so for these also. Kind of a pain, but I don't want to end up w/ some unexplained B4B difference in the future. I can start that now, since it depends only on the #945 PR (ie, it doesn't depend on the input-data being staged). 

As for the input-data, yes. That should be the same on all platforms.

DeniseWorthen committed:
Merge branch 'feature/wts4prep' into feature/ocnprep

DeniseWorthen commented:
The new weights can be copied from these locations:

orion: ``/work/noaa/stmp/dworthen/CPLD_GRIDGEN/rt_243218``
hercules: ``/work/noaa/stmp/dworthen/CPLD_GRIDGEN/rt_1793309``
jet: ``/lfs4/HFIP/h-nems/Denise.Worthen/CPLD_GRIDGEN/rt_356270``
wcoss2: ``/lfs/h2/emc/stmp/denise.worthen/CPLD_GRIDGEN/rt_259430``

GeorgeGayno-NOAA commented on CMakeLists.txt:
```diff
@@ -69,6 +71,7 @@ if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -traceback")
   set(CMAKE_Fortran_FLAGS_RELEASE "-O3 -fp-model precise")
   set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -check all -ftrapuv")
+  #set(CMAKE_Fortran_FLAGS_DEBUG "-check bounds -fpe0 -ftrapuv -fPIC -init=snan,arrays -check uninit")
```
Don't forget to remove this when you are done testing.

DeniseWorthen committed:
Merge branch 'ufs-community:develop' into feature/ocnprep

DeniseWorthen committed:
remove debug setting used during testing

DeniseWorthen committed:
remove changes in build_all.sh during testing

DeniseWorthen committed:
update unit test nmls for generic location

DeniseWorthen commented:
@GeorgeGayno-NOAA If all the weights and input data have been staged, do you want me to create baselines for this PR?

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA If all the weights and input data have been staged, do you want me to create baselines for this PR?

The file whacker removed your weights files on WCOSS2 before I could host them. Sorry. Can you please regenerate?

You can create your baselines on the RDHPCS machines. Thanks.

DeniseWorthen commented:
@GeorgeGayno-NOAA Did you get the new weights files in place on jet? I'm getting a failure 

```
FATAL ERROR: open: /lfs4/HFIP/hfv3gfs/emc.nemspara/role.ufsutils/ufs_utils/reg_tests/cpld_gridgen/baseline_data/050/tripole.mx050.Ct.to.Cu.bilinear.nc : No such file or directory
```

DeniseWorthen commented:
On WCOSS2, the new weights can be found at ``/lfs/h2/emc/stmp/denise.worthen/CPLD_GRIDGEN/rt_189836/``

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA Did you get the new weights files in place on jet? I'm getting a failure
> 
> ```
> FATAL ERROR: open: /lfs4/HFIP/hfv3gfs/emc.nemspara/role.ufsutils/ufs_utils/reg_tests/cpld_gridgen/baseline_data/050/tripole.mx050.Ct.to.Cu.bilinear.nc : No such file or directory
> ```

Sorry. Try it now.

DeniseWorthen committed:
switch to /work2 on hercules

* use /work2 for cpld_grdgen and ocnprep on hercules. orion will
use /work

DeniseWorthen commented:
@GeorgeGayno-NOAA One thing I just noticed is that the RT, at least for ``cpld_gridgen`` and ``ocnice_prep`` both use the same ``stmp`` space on hercules and orion, so I switched both regtests to ``work2`` for hercules.
 

Here are the ocnprep baselines:

jet: ``/lfs4/HFIP/h-nems/Denise.Worthen/OCNICE_PREP/BASELINE``
hera: ``/scratch1/NCEPDEV/stmp4/Denise.Worthen/OCNICE_PREP/BASELINE``
orion: ``/work/noaa/stmp/dworthen/OCNICE_PREP/BASELINE``
hercules: ``/work2/noaa/stmp/dworthen/OCNICE_PREP/BASELINE``
wcoss2: ``/lfs/h2/emc/stmp/denise.worthen/OCNICE_PREP/BASELINE``

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA One thing I just noticed is that the RT, at least for `cpld_gridgen` and `ocnice_prep` both use the same `stmp` space on hercules and orion, so I switched both regtests to `work2` for hercules.
> 
> Here are the ocnprep baselines:
> 
> jet: `/lfs4/HFIP/h-nems/Denise.Worthen/OCNICE_PREP/BASELINE` hera: `/scratch1/NCEPDEV/stmp4/Denise.Worthen/OCNICE_PREP/BASELINE` orion: `/work/noaa/stmp/dworthen/OCNICE_PREP/BASELINE` hercules: `/work2/noaa/stmp/dworthen/OCNICE_PREP/BASELINE` wcoss2: `/lfs/h2/emc/stmp/denise.worthen/OCNICE_PREP/BASELINE`

I hosted the baseline files in their official locations.

DeniseWorthen committed:
update memory for wcoss2

DeniseWorthen committed:
add WCOSS2 log

DeniseWorthen committed:
add hercules log

DeniseWorthen committed:
add orion log

DeniseWorthen committed:
add hera log

DeniseWorthen commented:
@GeorgeGayno-NOAA Everything is ready except for Jet. I'll need to wait until the issues w/ lfs4 are fixed.

GeorgeGayno-NOAA commented on sorc/ocnice_prep.fd/ocniceprep.nml:
These namelist files are not source code. Can they be moved under a 'parm' directory?

GeorgeGayno-NOAA commented on sorc/ocnice_prep.fd/restarts_mod.F90:
This module needs doxygen.

DeniseWorthen replied:
I've added the doxygen. This surprised me, because I thought the build docs=on would fail if there wasn't doxygen present. Thanks for catching it.

DeniseWorthen committed:
remove namelist files

* namelist is located in reg_tests/ocniceprep/parm

DeniseWorthen committed:
add doxygen to restarts_mod

GeorgeGayno-NOAA commented on sorc/ocnice_prep.fd/CMakeLists.txt:
```diff
@@ -0,0 +1,49 @@
+  esmf
+  MPI::MPI_Fortran
+  NetCDF::NetCDF_Fortran)
+
```
Is this program threaded? I am able to build without the OpenMP library.

DeniseWorthen replied:
No, this code not threaded. Should I remove this?

GeorgeGayno-NOAA replied:
Yes.

GeorgeGayno-NOAA replied:
The OpenMP library, not the MPI library. I would think that MPI is needed because of ESMF.

DeniseWorthen replied:
You know, I thought about that! But I see now the comment was supposed to be for the next line and I didn't expand the comment. Let me fix it.

GeorgeGayno-NOAA replied:
Looks the MPI library gets mapped in regardless of the MPI::MPI_Fortran line. You can remove that line as well as the OpenMP line.

GeorgeGayno-NOAA commented on ush/ocnice_prep.sh:
```diff
@@ -0,0 +1,55 @@
+fi
+if [ $RESNAME = 100 ]; then
+    export DSTDIMS="360,320"
+fi
```
Can you remove the reference to your personal directory?

DeniseWorthen replied:
This actually came in when I built the RT by modifying the cpld_gridgen RT, so something similar is present there. Let me see if I can fix both.

DeniseWorthen committed:
remove unneeded items

* remove MPI from CMakeLists.txt and hardwired outdir settings
 in cpld_gridgen and ocnice_prep RT scripts

DeniseWorthen committed:
update CMakeLists.txt

GeorgeGayno-NOAA commented:
@DeniseWorthen - one more thing before merging. The `./reg_tests/rt.sh` script needs to be updated for the new test. I have made the update and tested on Hera. Please find it here: `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS.denise/reg_tests`

DeniseWorthen committed:
update reg_tests/rt.sh script

DeniseWorthen committed:
add jet log

GeorgeGayno-NOAA reviewed (approved):
Looks good. Will merge.
